devices/disk-bios/disk-bios.c

```diff
@@ -285,8 +280,7 @@ static ssize_t diskbios_write(unsigned int minor, addr_t offs, const void *buff,
 
 		/* Write data to cache */
 		size = (sb == eb) ? len - l : SIZE_BLOCK - offs % SIZE_BLOCK;
-		hal_memcpy(diskbios_common.wcache + b * SIZE_BLOCK + offs % SIZE_BLOCK, buff, size);
-		buff += size;
+		hal_memcpy(wcache + b * SIZE_BLOCK + offs % SIZE_BLOCK, (const char *)buff + l, size);
```

Comment:
I think that brackets are needed here - intended operation order isn't clear

Reply:
Fixed.